src/test/java/cat/udl/eps/softarch/academicrecruit/steps/CreateDocumentStepDefs.java

```diff
@@ -14,17 +16,23 @@
 public class CreateDocumentStepDefs {
     final StepDefs stepDefs;
     final DocumentRepository documentRepository;
+    final CandidateRepository candidateRepository;
+
     final SelectionProcessRepository selectionProcessRepository;
     private String newResourceUri;
-    public CreateDocumentStepDefs(StepDefs stepDefs, DocumentRepository documentRepository, SelectionProcessRepository selectionProcessRepository) {
+    public CreateDocumentStepDefs(StepDefs stepDefs, DocumentRepository documentRepository, SelectionProcessRepository selectionProcessRepository,  CandidateRepository candidateRepository) {
         this.stepDefs = stepDefs;
         this.documentRepository = documentRepository;
         this.selectionProcessRepository = selectionProcessRepository;
+        this.candidateRepository = candidateRepository;
+
     }
     @When("I create a document with title {string}")
     public void iCreateANewDocumentWithTitle(String title) throws Throwable {
         Document document = new Document();
         document.setTitle(title);
+        documentRepository.save(document);
```

Comment:
If you store the Document here, you can't check if the Document is correctly created through the POST you do below.